lib/vec.c

```diff
@@ -178,13 +186,19 @@ static int bufvec_alloc(struct m0_bufvec *bufvec,
 
 			if (bufvec->ov_buf[i] == NULL)
 				goto fail;
+			if (M0_FI_ENABLED("buf-alloc-fail"))
+				goto fail;
```

Comment:
If this FI is triggered in the middle of this loop, the function will fail, and the the error handling can still cleanup the allocated memory, right?

Reply:
Yes, exactly.